efi/pe.go

```diff
@@ -0,0 +1,261 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2023 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package efi
+
+import (
+	"bytes"
+	"crypto"
+	"encoding/csv"
+	"errors"
+	"fmt"
+	"io"
+	"strconv"
+
+	efi "github.com/canonical/go-efilib"
+	"golang.org/x/xerrors"
+
+	pe "github.com/snapcore/secboot/internal/pe1.14"
+)
+
+type sbatComponent struct {
+	Name              string
+	Generation        int
+	VendorName        string
+	VendorPackageName string
+	VendorVersion     string
+	VendorUrl         string
+}
+
+// peImageHandle provides utilities for working with a PE image.
```

Comment:
with a PE image's sections and signatures.

Reply:
I've updated this now.